hack/ci-e2e.sh

```diff
@@ -118,29 +118,19 @@ IRONIC_OVERLAY="${REPO_ROOT}/ironic-deployment/overlays/e2e"
 
 IRONIC_USERNAME="$(tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 12 | head -n 1)"
 IRONIC_PASSWORD="$(tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 12 | head -n 1)"
-IRONIC_INSPECTOR_USERNAME="$(tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 12 | head -n 1)"
-IRONIC_INSPECTOR_PASSWORD="$(tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 12 | head -n 1)"
 
 echo "${IRONIC_USERNAME}" > "${BMO_OVERLAY}/ironic-username"
 echo "${IRONIC_PASSWORD}" > "${BMO_OVERLAY}/ironic-password"
-echo "${IRONIC_INSPECTOR_USERNAME}" > "${BMO_OVERLAY}/ironic-inspector-username"
-echo "${IRONIC_INSPECTOR_PASSWORD}" > "${BMO_OVERLAY}/ironic-inspector-password"
 
 BMO_UPGRADE_FROM_OVERLAY="${REPO_ROOT}/config/overlays/e2e-release-0.4"
 echo "${IRONIC_USERNAME}" > "${BMO_UPGRADE_FROM_OVERLAY}/ironic-username"
 echo "${IRONIC_PASSWORD}" > "${BMO_UPGRADE_FROM_OVERLAY}/ironic-password"
-echo "${IRONIC_INSPECTOR_USERNAME}" > "${BMO_UPGRADE_FROM_OVERLAY}/ironic-inspector-username"
-echo "${IRONIC_INSPECTOR_PASSWORD}" > "${BMO_UPGRADE_FROM_OVERLAY}/ironic-inspector-password"
```

Comment:
This part is still needed since we have inspector in release-0.4

Reply:
I was wrong. The issue was not what I thought.

For reference, the BMO e2e tests currently work like this
- Ironic is deployed from [this](https://github.com/metal3-io/baremetal-operator/blob/main/ironic-deployment/overlays/e2e/kustomization.yaml), which is using [this ironic manifest](https://github.com/metal3-io/baremetal-operator/blob/main/ironic-deployment/base/ironic.yaml) that has the image `quay.io/metal3-io/ironic` (latest).
- BMO is deployed from [this](https://github.com/metal3-io/baremetal-operator/blob/main/config/overlays/e2e/kustomization.yaml) normally, which is using a freshly built BMO image every time.
- BMO is deployed from [this](https://github.com/metal3-io/baremetal-operator/blob/main/config/overlays/e2e-release-0.4/kustomization.yaml) for upgrade tests, which is using `quay.io/metal3-io/baremetal-operator:release-0.4`.

I saw the optional test fail with 

```
lstat /home/metal3ci/slave_root/workspace/metal3-bmo-e2e-test-optional-pull/config/overlays/e2e-release-0.4/ironic-inspector-username: no such file or directory"
```

and thought that we still had inspector there, but of course the test is just upgrading BMO, Ironic stays the same throughout.

So the *actual* reason it failed is just that the BMO e2e release-0.4 overlay is expecting the files to be there. You just have to remove this:

https://github.com/metal3-io/baremetal-operator/blob/ab253c0f3ee643b9b7a883c0b0b4cc3a4485bbd6/config/overlays/e2e-release-0.4/kustomization.yaml#L41-L44

Reply:
Yeah, good point. But that only works with BMO 0.5 and newer, the previous versions require Inspector.

Reply:
Ok I think I understand. So for release-0.4 we would need to deploy inspector and that means we need a separate ironic overlay. We can do that in the same way as the BMO release-0.4 overlay, i.e. point to the release branch where we still have inspector.
https://github.com/metal3-io/baremetal-operator/blob/3cd8e3871eb623a47a835904bd7d4cf4b0073707/config/overlays/e2e-release-0.4/kustomization.yaml#L5

Reply:
CC @mquhuy this is relevant for you also if you start working on ironic upgrades.

Reply:
So, which upgrades do we test now on master? If only from 0.5, we can proceed with your earlier suggestion.

Reply:
We are actually planning to have multiple BMO upgrade tests as in #1451 . Is it possible for `release-0.5` and BMO `main` to use `Inspector` still? In that case I think it would be easier to just keep `inspector` for upgrade tests, until we no longer support release-0.3 and release-0.4

Reply:
@mquhuy we should test going from having inspector to not having it. That is the most valuable test!
For now we do not touch Ironic so all whatever we deploy to start with is what you use also after BMO upgrade, that means we will have to use inspector for at least upgrades from 0.4 and 0.3. I think the way forward is to create one overlay for ironic with inspector and one without inspector. For release-0.5 upgrade we can use the one without inspector then.

Reply:
@dtantsur yes for now we only upgrade from 0.5 so you can completely remove inspector. (It was 0.4 until recently which is probably why the test failed on this PR. If you rebase I think it will be fine.)

Reply:
Yes, it makes sense